### PR TITLE
chore(EMI-2082): refactor collector signals on Artwork details screen

### PR DESCRIPTION
**Pull request by oxaudo**
This PR resolves [EMI-2082] 

> [!NOTE]
> Part of this work is hidden behind `AREnableCuratorsPicksAndInterestSignals` feature flag

### Description
This PR refactors Increased Interest and Curator's Pick and Showing Now signals on artwork details page to show only one signal at a time based on signal data and feature flag presense

### Screenshots
| | Before | After |
|---|---|---|
| Artwork screen | Screen Shot 2024-09-16 at 1 50 01 PM  |  Screen Shot 2024-09-16 at 3 11 54 PM

### PR Checklist

- [ ] I have tested my changes on **iOS** and **Android**.
- [X] I hid my changes behind a **[feature flag]**, or they don't need one.
- [X] I have included **screenshots** or **videos**, or I have not changed the UI.
- [X] I have added **tests**, or my changes don't require any.
- [X] I added an **[app state migration]**, or my changes do not require one.
- [ ] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [X] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- consolidated Interest, Curator's Pick and Showing Now signals to be shown one at a time (hidden behind FF) - oxaudo

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

cc @artsy/emerald-devs 

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[EMI-2082]: https://artsyproduct.atlassian.net/browse/EMI-2082?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Cross-platform user-facing changes (consolidated Interest, Curator's Pick and Showing Now signals to be shown one at a time (hidden behind FF) - oxaudo)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 15b14a0618eee56f3f374916d78ae27c4cb43d08
</p>

File: src/app/Scenes/Artwork/Components/ArtworkCollectorSignalsRow.tsx [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+  const { collectorSignals } = useFragment(fragment, artwork)
+  if (!collectorSignals) {
+    return null
+  }
+
+  const { curatorsPick, increasedInterest, runningShow } = collectorSignals
+  if (!curatorsPick && !increasedInterest && !runningShow) {
+    return null
+  }
```
Comment (MrSltun) [inline src/app/Scenes/Artwork/Components/ArtworkCollectorSignalsRow.tsx:34]: **Suggestion**: can we combine these to be one check? something like this?

```suggestion
  if (!collectorSignals || (!collectorSignals.curatorsPick && !collectorSignals.increasedInterest && !collectorSignals.runningShow)) {
    return null
  }

```

File: src/app/Scenes/Artwork/Components/ArtworkCollectorSignalsRow.tsx [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+      signalDescription = "Based on collector activity in the past 14 days"
+      SignalIcon = TrendingIcon
+      break
+    }
+    case !!runningShow: {
+      if (
+        !runningShow?.startAt ||
+        !runningShow?.endAt ||
+        !runningShow?.href ||
+        !runningShow?.name
```
Comment (MrSltun) [inline src/app/Scenes/Artwork/Components/ArtworkCollectorSignalsRow.tsx:59]: **Question**: I see that the show here requires a link, but in the previous component, it was either a link or a `#` ([code](https://github.com/artsy/eigen/blob/main/src/app/Scenes/Artwork/Components/ArtworkShowingNowCollectorSignal.tsx#L35)), was this intentional?

File: src/app/Scenes/Artwork/Components/ArtworkCollectorSignalsRow.tsx [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+    <Flex flexDirection="row" pt={4} pb={2}>
+      <SignalIcon mr={0.5} fill="black60" height={25} width={25} />
+
+      <Flex flexDirection="column">
+        <Text variant="sm-display" color="black100">
+          {singalTitle}
+        </Text>
+
+        {href ? (
+          <LinkText variant="sm" color="black60" onPress={() => navigate(href || "#")}>
```
Comment (MrSltun) [inline src/app/Scenes/Artwork/Components/ArtworkCollectorSignalsRow.tsx:85]: **Suggestion**: If we know that `href` exists, maybe we should remove the logical OR in the `navigate` function?

File: src/app/Scenes/Artwork/Components/ArtworkCollectorSignalsRow.tests.tsx [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+              name: "Saenger Galería at The Armory Show 2024",
+              href: "/show/saenger-galeria-saenger-galeria-at-the-armory-show-2024-1",
+              startAt: "2024-08-28T00:00:00+00:00",
+              endAt: "2024-09-22T00:00:00+00:00",
+            },
+          },
+        }),
+      })
+
+      expect(screen.queryByText(/Showing now/)).not.toBeOnTheScreen()
```
[the remark this answers is not in the document]
Comment (oxaudo) [reply src/app/Scenes/Artwork/Components/ArtworkCollectorSignalsRow.tests.tsx:139]: The one below `renders the curator's pick signal even when there's an increased interest signal` believe is testing that, no? Or you mean something different?

File: src/app/Scenes/Artwork/Components/ArtworkCollectorSignalsRow.tsx (hunk 1 above)
Comment (oxaudo) [reply src/app/Scenes/Artwork/Components/ArtworkCollectorSignalsRow.tsx:34]: Totally could. In my mind the other one is more readable. Like if collector signals are not present - nothing is rendered. If they are present - start analyzing them... But maybe there is some performance win in doing it togethere?

File: src/app/Scenes/Artwork/Components/ArtworkCollectorSignalsRow.tsx (hunk 3 above)
Comment (oxaudo) [reply src/app/Scenes/Artwork/Components/ArtworkCollectorSignalsRow.tsx:85]: I was getting a typescript error about possible type mismatch if I did not include this `#` which is strange to me as the code above supposedly tests that it's present. Not sure why typescript does not account for that in this case :(

File: src/app/Scenes/Artwork/Components/ArtworkCollectorSignalsRow.tsx (hunk 2 above)
Comment (oxaudo) [reply src/app/Scenes/Artwork/Components/ArtworkCollectorSignalsRow.tsx:59]: Yea, I was hoping to not need to do `|| "#"` but that did not work at the end :(

File: src/app/Scenes/Artwork/Components/ArtworkCollectorSignalsRow.tsx (hunk 1 above)
Comment (MrSltun) [reply src/app/Scenes/Artwork/Components/ArtworkCollectorSignalsRow.tsx:34]: I believe yes since there's no need to destructuring the `collectorSignals` object if the they're all false

File: src/app/Scenes/Artwork/Components/ArtworkCollectorSignalsRow.tsx (hunk 2 above)
Comment (MrSltun) [reply src/app/Scenes/Artwork/Components/ArtworkCollectorSignalsRow.tsx:59]: so if the show doesn't have a link, it won't show up, right?

File: src/app/Scenes/Artwork/Components/ArtworkCollectorSignalsRow.tsx (hunk 3 above)
Comment (MrSltun) [reply src/app/Scenes/Artwork/Components/ArtworkCollectorSignalsRow.tsx:85]: It does account for it, and removing `|| '#'` doesn't show an error or a warning, I just tested it locally 👀

File: src/app/Scenes/Artwork/Artwork.tsx [hunk 5]
```diff
@@ -308,19 +307,8 @@ export const Artwork: React.FC<ArtworkProps> = (props) => {
...
-        })
-      }
-
-      if (enableCuratorsPicksAndInterestSignals) {
-        sections.push({
-          key: "artworkCuratorsPickIncreasedInterestCollectorSignal",
-          element: (
-            <ArtworkCuratorsPickIncreasedInterestCollectorSignal artwork={artworkAboveTheFold} />
-          ),
+          key: "artworkCollectorSignalsRow",
```
[the remark this answers is not in the document]
Comment (MrSltun) [reply src/app/Scenes/Artwork/Artwork.tsx:307]: I like your idea, I think `ArtworkDetailsCollectorSignal` is a good one

File: src/app/Scenes/Artwork/Components/ArtworkCollectorSignalsRow.tsx (hunk 3 above)
Comment (oxaudo) [reply src/app/Scenes/Artwork/Components/ArtworkCollectorSignalsRow.tsx:85]: Yea, locally it compiles for me as well but when I try to push to branch get this error:
Screen Shot 2024-09-17 at 12 43 23 PM

I can force push but that seems wrong @MrSltun ?

File: ios/Podfile.lock
Comment (MrSltun) [inline ios/Podfile.lock:1]: **Question**: Is this change intentional?

**Review by MrSltun — APPROVED** [review]
Looks good! I have only one question about `Podfile.lock` change, and if possible let's remove it?

File: ios/Podfile.lock
Comment (oxaudo) [reply ios/Podfile.lock:1]: Oh... No:( I need to revert it. That is something that I need to figure out. I keep getting this state when I rebuild and I need to remember to ignore it. Here I did not :(

**Comment by oxaudo** [comment]
Ok. Reverted podfile (thinks for noticing!). Will merge when green.

File: src/app/Scenes/Artwork/Components/ArtworkDetailsCollectorSignal.tsx [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+  let singalTitle: string | null = null
+  let signalDescription: string | null = null
+  let href: string | null = null
+  let SignalIcon = FairIcon
+
+  switch (true) {
+    case curatorsPick && enableCuratorsPicksAndInterestSignals: {
+      singalTitle = "Curators’ Pick"
+      signalDescription = "Hand selected by Artsy curators this week"
+      SignalIcon = VerifiedIcon
```
Comment (MounirDhahri) [inline src/app/Scenes/Artwork/Components/ArtworkDetailsCollectorSignal.tsx:48]: **suggestion:** maybe this is a bit late - but it would have been great if we were getting these values from MP

Comment (oxaudo) [reply src/app/Scenes/Artwork/Components/ArtworkDetailsCollectorSignal.tsx:48]: You mean the name of the icon for each signal should come from MP? This is an interesting idea for refactor. The icon does feel very UI specific but do see your point that if it should be the same across platforms might make sense to keep it in MP. I'll bring it up to the team and see if we want to go with this suggestion. Will create a separate card for it:)
